web/html/src/manager/notifications/notification-messages.js

```diff
@@ -222,23 +218,33 @@ class NotificationMessages extends React.Component {
     return Object.keys(message).map((id) => message[id]);
   };
 
-  buildDescription = (row) => {
-    let description = null;
-    switch(row['type']) {
-      case 'OnboardingFailed':
-        description = 'Error registering minion id: ' + row['data']['minionId'];
-      break;
-      case 'ChannelSyncFailed':
-        description = <span>Error syncing the channel: <a href={"/rhn/channels/ChannelDetail.do?cid=" + row['data']['channelId']}>{row['data']['channelName']}</a></span>;
-      break;
-      case 'ChannelSyncFinished':
-        description = <span>Channel <a href={"/rhn/channels/ChannelDetail.do?cid=" + row['data']['channelId']}>{row['data']['channelName']}</a> sync completed</span>;
-      break;
-      default: description = JSON.stringify(row['data']);
-    }
-    return description;
+  showDescriptionPopup = (row) => {
+    this.setState({popupItem: row});
+    showDialog("notifications-popup-dialog");
+  }
+
+  buildSummary = (row) => {
+    const popupLink = <a href="#" onClick={() => this.showDescriptionPopup(row)}>{t("Details")}</a>;
```

Comment:
Is this link going to be always visible? Maybe, depending on the type of the message, if the `description` field does not deliver any additional value to the `summary` maybe we can hide it. I wonder if it could be the case to already define this on the backend side, leaving the `description` field `null` or `String.empty` and we can decode it on the frontend side. Just my 2c

Reply:
We already do that in backend. If the description is absent, frontend receives `undefined` value. And the logic on line 233 "hides" the link (this happens on all "falsy" values: `null`, `undefined`, empty string, ...).

Reply:
Ups, thanks for clarifying it. I probably overlooked the logic, sorry for the noise. :+1: